sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -369,6 +373,19 @@ def tune(
                         f"Incorrect value for env_per_trial: {env_per_trial}"
                     )
 
+        # Add metrics collector to the Katib Experiment.
+        # Up to now, We only support parameter `kind`, of which default value is `StdOut`, to specify the kind of metrics collector. 
+        experiment.spec.metrics_collector = models.V1beta1MetricsCollectorSpec(
+            collector=models.V1beta1CollectorSpec(kind=metrics_collector_config["kind"])
+        )
+        if metrics_collector_config["kind"] == "Push":
+            trial_params.append(
+                models.V1beta1TrialParameterSpec(name="trialName", reference="${{trialSpec.Name}}")
+            )
+            env.append(
+                client.V1EnvVar(name="KATIB_TRIAL_NAME", value="${{trialParameters.trialName}}")
+            )
+
```

Comment:
Do we want to always pass KATIB_TRIAL_NAME to all Katib Trials ?
I feel that this parameter is very useful for different use-cases and we should always pass it to the training container.
For example, when user wants to export trained model and model prefix = Trial name.

Reply:
Yeah, it seems that it's worthwhile to always pass `KATIB_TRIAL_NAME` to Katib Trials. 

I'll modify the code and add some comment in the doc string.

Reply:
@Electronic-Waste What if we use `fieldPath` for `KATIB_TRIAL_NAME` env as follows:
```
name: RANK
valueFrom:
  fieldRef:
    fieldPath: metadata.labels['katib.kubeflow.org/trial']
```

Like in this [JobSet example](https://github.com/kubernetes-sigs/jobset/blob/main/examples/pytorch/cnn-mnist/mnist.yaml#L27-L30).

I believe, we already mutate Trial name [to the Trial's Pod labels.](https://github.com/kubeflow/katib/blob/master/pkg/webhook/v1beta1/pod/utils.go#L278).

Reply:
@andreyvelich Okay, I see. I'll commit new changes to this branch soon.